apm-agent-plugins/apm-jmx-plugin/src/main/java/co/elastic/apm/agent/jmx/JmxMetricTracker.java

```diff
@@ -186,20 +192,28 @@ private void registerMBeanNotificationListener(final MBeanServer server) {
         listener = new NotificationListener() {
             @Override
             public void handleNotification(Notification notification, Object handback) {
-                if (notification instanceof MBeanServerNotification) {
-                    ObjectName mBeanName = ((MBeanServerNotification) notification).getMBeanName();
-                    for (JmxMetric jmxMetric : jmxConfiguration.getCaptureJmxMetrics().get()) {
-                        if (jmxMetric.getObjectName().apply(mBeanName)) {
-                            logger.debug("MBean added at runtime: {}", jmxMetric.getObjectName());
-                            register(Collections.singletonList(jmxMetric), server);
-                        }
+                try {
+                    if (notification instanceof MBeanServerNotification) {
+                        onMBeanAdded(((MBeanServerNotification) notification).getMBeanName());
+                    }
+                } catch (Exception e) {
+                    logger.error(e.getMessage(), e);
+                }
+            }
+
+            private void onMBeanAdded(ObjectName mBeanName) {
+                logger.trace("Receiving MBean registration notification for {}", mBeanName);
+                for (JmxMetric jmxMetric : jmxConfiguration.getCaptureJmxMetrics().get()) {
+                    if (jmxMetric.getObjectName().apply(mBeanName)) {
+                        logger.debug("MBean added at runtime: {}", jmxMetric.getObjectName());
+                        register(Collections.singletonList(jmxMetric), server);
                     }
                 }
             }
         };
         try {
             server.addNotificationListener(MBeanServerDelegate.DELEGATE_NAME, listener, filter, null);
-        } catch (InstanceNotFoundException e) {
+        } catch (Exception e) {
```

Comment:
why do we need to catch such a broad exception ? Are there common NPE or IllegalStateException thrown ?

Reply:
No, didn't see any of those in the wild but if there are any I think we should still continue and try monitor already registered MBeans than to give up jmx monitoring entirely.

In theory, this could happen if there's a customer MBean server that does not allow to add notification listeners.